app/views/articles/_actions.html.erb

```diff
@@ -5,21 +5,21 @@
       <%= render partial: "articles/reaction_button",
                  locals: {
                    category: :like,
-                   description: "Heart",
+                   description: "Like post/unlike post",
```

Comment:
refactor (blocking) - Changing the description here is helpful but unfortunately it's not giving enough context for screen reader users, as there's no indication of the current state - e.g. If I click this button is it going to like or unlike the post? 🙃 

I took a little look into it and I think actually the simplest way to resolve this will be a different approach:

- Revert the changes to these descriptions (they're not going to be used to create the button name any more, but they do still provide the popover description on mouse hover)
- In `_reaction_button.html.erb`, for each svg update to remove the `title` attribute and use `aria-hidden` (e.g. `<%= inline_svg_tag(image_path, aria_hidden: true, class: "crayons-icon") %>`. This is going to hide the icon information from screen reader users, as it's not helpful
- In `initializeArticleReactions.js`, we can add the appropriate `aria-label` to each reaction button depending on whether or not it's selected (see below)

I would suggest achieving point 3 by adding a helper function to retrieve the right text, e.g:

```js
function getReactionAriaLabel(reactionName, reacted) {
  switch (reactionName) {
    case 'readinglist':
      return reacted ? 'Remove from reading list' : 'Add to reading list';
    case 'unicorn':
      return reacted ? 'Remove unicorn reaction' : 'React with unicorn';
    case 'like':
      return reacted ? 'Unlike' : 'Like';
  }
}
```

And then calling this in the two methods `showUserReaction` and `hideUserReaction`:

```js
function showUserReaction(reactionName, animatedClass) {
  const reactionButton = document.getElementById(
    'reaction-butt-' + reactionName,
  );
  reactionButton.classList.add('user-activated', animatedClass);
  reactionButton.setAttribute(
    'aria-label',
    getReactionAriaLabel(reactionName, true),
  );
}

function hideUserReaction(reactionName) {
  const reactionButton = document.getElementById(
    'reaction-butt-' + reactionName,
  );
  reactionButton.classList.remove('user-activated', 'user-animated');
  reactionButton.setAttribute(
    'aria-label',
    getReactionAriaLabel(reactionName, false),
  );
}
```

NB: the lines in those functions there is also a line adding/removing 'aria-checked' which needs deleted too, since they are no longer checkboxes

Let me know if you need any assistance with any of this! It's going to make the user experience a lot better for screen reader users so thank you for looking at it 🙂

Reply:
looks better, will add this, thanks for review.